MdePkg/Library/BaseRngLib/AArch64/Rndr.c

```diff
@@ -43,15 +38,12 @@ BaseRngLibConstructor (
   VOID
   )
 {
-  UINT64  Isar0;
-
   //
   // Determine RNDR support by examining bits 63:60 of the ISAR0 register returned by
   // MSR. A non-zero value indicates that the processor supports the RNDR instruction.
   //
-  Isar0 = ArmReadIdIsar0 ();
-
-  mRndrSupported = ((Isar0 & RNDR_MASK) != 0);
+  mRndrSupported = !((ArmReadIdAA64Isar0Reg () >> ARM_ID_AA64ISAR0_EL1_RNDR_SHIFT) &
+                     ARM_ID_AA64ISAR0_EL1_RNDR_MASK);

```

Comment:
Sorry for not spotting this right way, but I think the negation is also wrong.

Shall we just keep the old code instead? I.e.,

```c
Isar0          = ArmReadIdAA64Isar0Reg ();
mRndrSupported = ((Isar0 & ARM_ID_AA64ISAR0_EL1_RNDR_MASK) != 0);
```

Reply:
My mistake, I should have updated the code to something correct.